tests/system_tests_cert_rotation.py

```diff
@@ -0,0 +1,199 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+"""
+Tests the routers TLS Certificate Rotation feature.
+"""
+
+import time
+from system_test import TestCase, main_module, Qdrouterd, unittest, retry
+from system_test import CA_CERT, SSL_PROFILE_TYPE, CONNECTION_TYPE
+from system_test import CLIENT_CERTIFICATE, CLIENT_PRIVATE_KEY, CLIENT_PRIVATE_KEY_PASSWORD
+from system_test import SERVER_CERTIFICATE, SERVER_PRIVATE_KEY, SERVER_PRIVATE_KEY_PASSWORD
+
+
+class InterRouterCertRotationTest(TestCase):
+    """
+    Validate the Certificate Rotation feature against interior inter-router connections.
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(InterRouterCertRotationTest, cls).setUpClass()
+
+    def router(self, name, test_config, data_connection_count, **kwargs):
+        config = [
+            ('router', {'mode': 'interior',
+                        'id': name,
+                        'dataConnectionCount': f"{data_connection_count}"}),
+            ('listener', {'port': self.tester.get_port(), 'role': 'normal'}),
+        ]
+        config.extend(test_config)
+        return self.tester.qdrouterd(name, Qdrouterd.Config(config), **kwargs)
+
+    def get_inter_router_conns(self, router):
+        mgmt = router.management
+        conns = mgmt.query(type=CONNECTION_TYPE).get_dicts()
+        return [c for c in conns if 'inter-router' in c['role']]
+
+    def wait_inter_router_conns(self, router, count):
```

Comment:
Seems generic and useful enough to move this function to `system_test.py`

Reply:
moved